### PR TITLE
Development

### PR DESCRIPTION
**Pull request by BenMukebo**
##  For this Rails Capstone project, here are what I did to implement that:

- Splash screen

  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages

  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)

  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page

  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
  - Create unit and integration tests for all the most important components of your RoR application.

- Make sure your app is tested adequately.
- Deploy the project, and test for final details.
- Record a video for your project.
- Create a good README and PR description.

### Live Demo

[Live Demo Link](https://limitless-savannah-05818.herokuapp.com/)

### Loom Video

[Loom Video Link](https://www.loom.com/share/2d1971886acf41eab27ea244c1f1acdb)

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+/* .home-header-categories h1 {
+  width: 100%;
+  line-height: 70px;
+  text-align: center;
+  font-size: 24px;
+  background-color: var(--main-color);
+  color: var(--on-secondary);
+} */
+
```
Comment (Emmyn5600) [inline app/assets/stylesheets/application.css]: - [ ] Kindly remove the commented code.

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  def index
+    @categories = current_user.categories
+  end
+
+  # GET /categories/1 or /categories/1.json
+  def show
+    # @category = set_category
+    @category = Category.find(params[:id])
+    puts "\n\n\n\n #{@category.records.size} \n\n\n\n"
+    @records = @category.records.order(created_at: :desc)
```
Comment (Emmyn5600) [inline app/controllers/categories_controller.rb]: - [ ] Remove the `puts` to print or console the value

File: app/controllers/records_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+class RecordsController < ApplicationController
+  before_action :set_record, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+  load_and_authorize_resource :category
+
+  def index
+    @records = Record.all
+    # @category = Category.find(params[:category_id])
+    # @records = @category.records
```
Comment (Emmyn5600) [inline app/controllers/records_controller.rb]: - [ ] Remove the commented code.

File: app/views/categories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<nav class="header-categories d-flex flex-center">
+  <%= link_to  destroy_user_session_path, class: "icons" do%>
+    <i class="fa-solid fa-toggle-on"></i>
+  <% end %>
+    <h1>CATEGORIES</h1>
+</nav>
+
```
Comment (Emmyn5600) [inline app/views/categories/index.html.erb]: - [ ] Kindly use the icon for `log out` or text for it to have a meaningful or descriptive message on our project.

image

File: app/views/categories/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,69 @@
+<% render @category %>
+
+<nav class="header-categories d-flex flex-center">
+  <div class="icons">
+    <%= link_to  categories_path do %>
+      <i class="fa-solid fa-right-to-bracket"></i>
+    <% end %>
+  </div>
+  <h1>CATEGORY DETAILS</h1>
+</nav>
```
Comment (Emmyn5600) [inline app/views/categories/show.html.erb]: - [ ] How about using the backward icon or other icons to show the back button.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @BenMukebo 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet, please work on it and ask for another review.

- You have a descriptive PR
- You have the descriptive Readme file.

 ## Required Changes ♻️
 
 - Check the comment under the review too.
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/assets/stylesheets/application.css (hunk 1 above)
Comment (BenMukebo) [reply app/assets/stylesheets/application.css]: Yeah sure 😢 I've forgotten to remove it!!

File: app/controllers/records_controller.rb (hunk 3 above)
Comment (BenMukebo) [reply app/controllers/records_controller.rb]: Thank you!!

File: app/views/categories/index.html.erb (hunk 4 above)
Comment (BenMukebo) [reply app/views/categories/index.html.erb]: Yes sure, I'm going to change it thanks!!

File: app/views/categories/show.html.erb (hunk 5 above)
Comment (BenMukebo) [reply app/views/categories/show.html.erb]: Hi @Emmyn5600,
I have implemented all changes requested right
Thank you very much to contribute on my professionalism, 🙏🏻

**Review by uwadonat — APPROVED** [review]
Hi @[BenMukebo](https://github.com/BenMukebo), 
# Congratulations! 🎉

Your project is complete!  

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**